### ISSUE TITLE
Need to improve on the http return codes with OpenBMC Python Function

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#4794]
I was doing a `bmcreboot` and had a loop on the `bmcstate`.  We need to match the output of the Perl handling to ensure that we maintain the same debug info.... 

Here's the output of python: 
```
Mon Feb  5 15:00:54 EST 2018
Mon Feb  5 15:00:55 2018 mid05tor12cn02: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://172.11.139.2/login -d '{"data": ["root", "xxxxxx"]}'
Mon Feb  5 15:00:55 2018 mid05tor12cn13: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://172.11.139.13/login -d '{"data": ["root", "xxxxxx"]}'
Mon Feb  5 15:00:55 2018 mid05tor12cn05: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://172.11.139.5/login -d '{"data": ["root", "xxxxxx"]}'
Mon Feb  5 15:00:55 2018 mid05tor12cn15: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://172.11.139.15/login -d '{"data": ["root", "xxxxxx"]}'
Mon Feb  5 15:00:55 2018 mid05tor12cn16: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://172.11.139.16/login -d '{"data": ["root", "xxxxxx"]}'
Mon Feb  5 15:00:56 2018 mid05tor12cn15: [openbmc_debug] login Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn15: BMC NotReady
Mon Feb  5 15:00:57 2018 mid05tor12cn05: [openbmc_debug] login Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn05: BMC NotReady
Mon Feb  5 15:00:57 2018 mid05tor12cn02: [openbmc_debug] login Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn02: BMC NotReady
Mon Feb  5 15:00:58 2018 mid05tor12cn13: [openbmc_debug] login Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn13: BMC NotReady
Mon Feb  5 15:00:58 2018 mid05tor12cn16: [openbmc_debug] login Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn16: BMC NotReady
Mon Feb  5 15:00:58 EST 2018
```

Here's the output of Perl 

```
Mon Feb  5 15:01:44 EST 2018
Mon Feb  5 15:01:44 2018 OpenBMC: [openbmc_debug]
mid05tor12cn16: Error: [500] Login to BMC failed: 500 Can't connect to 172.11.139.16:443.
mid05tor12cn13: Error: [500] Login to BMC failed: 500 Can't connect to 172.11.139.13:443.
mid05tor12cn15: Error: [500] Login to BMC failed: 500 Can't connect to 172.11.139.15:443.
mid05tor12cn02: Error: [500] Login to BMC failed: 500 Can't connect to 172.11.139.2:443.
mid05tor12cn05: Error: [500] Login to BMC failed: 500 Can't connect to 172.11.139.5:443.
mid05tor12cn13: Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn16: Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn15: Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn02: Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn05: Error: BMC did not respond. Validate BMC configuration and retry the command.
Mon Feb  5 15:02:04 EST 2018
```

We need this same output message because the first one doesn't show us any value in a quick glance.  Even without debug turned on we need to know what is returned from the http server.

**Comment by xuweibj** [comment]
Hi @whowutwut , I think for command ``rpower bmcstate`` return "BMC NotReady" is the better output. Why we must make it same with perl. I thought we print out msg like perl is because it's hard to show "BMC NotReady" so print it like this. But in python, we could know it well that we could not connect BMC, so we give more accurate output.

**Comment by robin2008** [comment]
Actually `500` in perl for such case is not returned from BMC. At that case, the BMC is not alive, it cannot respond anything.

**Comment by whowutwut** [comment]
@xuweibj @robin2008 There was a lot of debug that I had to do at the end of Nov last year that was really painful because xCAT hide the real the error of the return code. 
 I could not easily tell the problem based on LBS copy and pasting me the output on the system . Resulted in at least this PR https://github.com/xcat2/xcat-core/pull/4407 . But there could be others... 

So this was done to help detect if there are BMC issues, not just say it's "Not Ready" because it doesn't allow us to know why.  There's so many reasons the bmc may not get into Ready state and one issue could be xCAT has a bug, or can not send the command successfully...  Even if the BMC is responding and ssh it up, it sometimes has some issues starting services...

So we really need to keep this logic because I spent a lot of time adding to the messages to be able to cut down on debug time, and if we do not have this, it will cause issues again in debug when switching to Python.

**Comment by robin2008** [comment]
Yes, I agree that.  But I just think it is better to log the details in logfile (/var/log/xcat/cluster.log) instead.

For python, if we need detail error information, we can do more checking when found error and log the corresponding error messages.

**Comment by whowutwut** [comment]
@robin2008 Not sure i agree, can you show me the example here of what is displayed? 
I've never looked in `/var/log/xcat/cluster.log` for issues when running an xCAT rpower command.. so it's not natural for me to look there.

**Comment by gurevichmark** [comment]
Do we still need this Issue ? Perhaps this is duplicate of #4855 and fixed by #4866 ?
